astropy/utils/introspection.py

```diff
@@ -1,23 +1,30 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 """Functions related to Python runtime introspection."""
 
+from __future__ import annotations
+
 import importlib
 import inspect
 import os
 import sys
 from importlib import metadata
 from importlib.metadata import packages_distributions
+from typing import TYPE_CHECKING
 
 from packaging.version import Version
 
 from .decorators import deprecated
 
+if TYPE_CHECKING:
+    from types import FrameType, ModuleType
```

Comment:
Weird that it fails even with `from __future__ import annotations`. Maybe quoting it would work ? (but you may also need noqa comments to prevent ruff from undoing it)

Reply:
RTD is complaining about `ModuleType`, but not about `FrameType` or `Literal`. That's making me think that it's the fact I've used `ModuleType` in a return annotation that's causing problems somehow.

Reply:
I'm not comfortable enough around RTD to say anything with confidence, but this smells like a potential bug on their side ?

Reply:
I tried a few different ways of importing `ModuleType`, but I couldn't get the documentation build to work properly. It seems that adding a new nitpick exception is the optimal solution under the circumstances.

Reply:
Yeah, RTD support is suboptimal